server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -685,6 +688,20 @@ protected Node(final Environment initialEnvironment,
         }
     }
 
+    /**
+     * @return A function that can be used to determine the requested REST compatible version
+     */
+    private CompatibleVersion getRestCompatibleFunction() {
+        List<RestCompatibilityPlugin> restCompatibilityPlugins = pluginsService.filterPlugins(RestCompatibilityPlugin.class);
+        CompatibleVersion compatibleVersion = CompatibleVersion.CURRENT_VERSION;
+        if (restCompatibilityPlugins.size() > 1) {
+            throw new IllegalStateException("Only one RestCompatibilityPlugin is allowed");
+        } else if (restCompatibilityPlugins.size() == 1) {
+            compatibleVersion = restCompatibilityPlugins.get(0)::getCompatibleVersion;
+        }
+        return compatibleVersion;
```

Comment:
This is a nit but I prefer making values final so it is easier to understand the assignment:

```suggestion
        final CompatibleVersion compatibleVersion;
        if (restCompatibilityPlugins.size() > 1) {
            throw new IllegalStateException("Only one RestCompatibilityPlugin is allowed");
        } else if (restCompatibilityPlugins.size() == 1) {
            compatibleVersion = restCompatibilityPlugins.get(0)::getCompatibleVersion;
        } else {
            compatibleVersion = CompatibleVersion.CURRENT_VERSION;
        }
        return compatibleVersion;
```